Makefile

```diff
@@ -19,7 +19,7 @@ endif
 JAVAFLAGS	= -XX:MaxPermSize=128M
 JAVACFLAGS  = -source 1.8 -target 1.8
 
-CPPFLAGS	+= -g0 -Os -Wall -Werror -Wno-long-long -std=c++11 -fno-addrsig
+CPPFLAGS	+= -g0 -O2 -Wall -Werror -Wno-long-long -std=c++11 -fno-addrsig
```

Comment:
@juliusmik Can you add a comment why performance optimization is needed here? So that in the future this flag is not toggled back and forth. If -Os produces smaller code size that's still good to know.